src/RPackage-Core/RPackageOrganizer.class.st

```diff
@@ -803,6 +783,15 @@ RPackageOrganizer >> removeCategory: aCategory [
 	SystemAnnouncer uniqueInstance classCategoryRemoved: category
 ]
 
+{ #category : #'system integration' }
+RPackageOrganizer >> removeClass: class [
+	"Remove the class, the class backpointer, the extensions and the extension backPointer from the receiver and the class involved with the class named: className. className is a class name and should not be a metaclass one. "
+
+	class isClassSide ifTrue: [ self error: 'We should only be able to remove classes and not instances of metaclass.' ].
```

Comment:
Does it mean that if we have only extension to a metaclass the package still get the instance side and the metaclass?

Reply:
I do not remember why I had this constraints about the metaclass.
Now I do not understand the `'We should only be able to remove classes and not instances of metaclass.'` because to me classes are instance of metaclass so I'm really confused.

Reply:
Maybe the comment is not good.
We can do `package removeClass: Object` but not `package removeClass: Object class`.

all of this instance side vs class side is because packages have caches based on instance or class side. I think I’ll try to unify that to remove all the checks in the code (but I’m doing one thing at a time right now)

Reply:
Ok this is just that you are aware of it.